moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -81,14 +132,21 @@ const NewProjectMap = ({
   refetchProjectDetails = null,
   noPadding = false,
   newFeature = false,
+  saveActionState = null,
+  saveActionDispatch = null,
+  componentEditorPanel = null,
 }) => {
   const classes = useStyles();
-  const mapRef = useRef();
+
   const featureCount = countFeatures(featureCollection);
   const selectedLayerIds = createSelectedIdsObjectFromFeatureCollection(
     featureCollection
   );
-  const mapControlContainerRef = useRef();
+
+  const mapRef = useRef();
+  const mapGeocoderContainerRef = useRef();
+  const mapEditToolsContainerRef = useRef();
+  const mapBasemapContainerRef = useRef();
```

Comment:
The refs looks sort of bad, but it was the only way I found to create portals and render map components outside the map.